mrmustard/math/backend_numpy.py

```diff
@@ -59,6 +59,13 @@ def __repr__(self) -> str:
     def abs(self, array: np.ndarray) -> np.ndarray:
         return np.abs(array)
 
+    def allclose(self, array1: np.array, array2: np.array, atol: float) -> bool:
```

Comment:
why is it necessary to reproduce this in the backend? There's no need for gradients

Reply:
For the tests, we needed an allclose that
- takes `atol`
- checks the shapes
Numpy's and tensorflow's don't do that. So we thought it best to add it to the backend, so we can use the same `allclose` in all the test files